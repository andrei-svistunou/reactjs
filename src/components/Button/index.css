.btn {
  color: #fff;
  padding: 7px 50px;
  border: none;
  border-radius: 2px;
}

.btn-primary {
  background-color: #f55263;
}

.btn--secondary {
  background-color: #444;
}

.btn--small {
  padding: 5px 20px;
  font-size: 12px;
}

.btn-invert {
  background: #fff;
  color: #f55263;
}
