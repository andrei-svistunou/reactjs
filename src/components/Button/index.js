import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Button = props => (
  <input type="button" className={`btn ${props.classes} upper-text`} value={props.value} onClick={props.onClick}/>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  classes: PropTypes.string.isRequired,
  onClick: PropTypes.func
};

export default Button;
