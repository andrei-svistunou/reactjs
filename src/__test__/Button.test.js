import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../components/Button';

describe('<Button />', () => {
  test('should render a button primary', () => {
    const btn = shallow(<Button classes='btn--primary' value='Click' onClick={jest.fn()}/>);

    expect(toJson(btn)).toMatchSnapshot();
  });

  test('should be get prop value', () => {
    const btn = shallow(<Button classes='btn--primary' value='Jest' onClick={jest.fn()} />);

    expect(btn.prop('value')).toEqual('Jest');
    expect(toJson(btn)).toMatchSnapshot();
  });
});
