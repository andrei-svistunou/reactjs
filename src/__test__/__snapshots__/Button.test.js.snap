// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be get prop value 1`] = `
<input
  className="btn btn--primary upper-text"
  onClick={[MockFunction]}
  type="button"
  value="Jest"
/>
`;

exports[`<Button /> should render a button primary 1`] = `
<input
  className="btn btn--primary upper-text"
  onClick={[MockFunction]}
  type="button"
  value="Click"
/>
`;
